scripts/auto-rebase/rebase.sh

```diff
@@ -36,13 +36,160 @@ RELEASE_TAG_RX="(.+)-([0-9]{4}-[0-9]{2}-[0-9]{2}-[0-9]{6})"
+    for arch in "${ARCHS[@]}"; do
+        write_lvms_images_for_arch "$arch"
+    done
+    popd
+}
+
+set_openshift_storage_ns(){
+        local manifest="$1"
+        local tmp="$(oc create --dry-run=client --namespace=openshift-storage -o yaml -f "${manifest}")"
+        # assume that if the returned string is nil, it's because the resource is not namespaced.  Failures
+        # like file-not-found will still result in the script exiting with a non-zero code.
+        if [ "$(echo "$tmp" | tr -d '[:space:]' )" -z ]; then
+            return 0
+        fi
+        echo "$tmp" > "$manifest"
+}
+
+update_lvms_manifests() {
+    local src="${STAGING_DIR}/lvms/amd64"
+    local dest="${REPOROOT}/assets/components/lvms"
+    local manifests=(
+          "topolvm-controller_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml"
+          "topolvm-controller_rbac.authorization.k8s.io_v1_clusterrole.yaml"
+          "topolvm-controller_v1_serviceaccount.yaml"
+          "topolvm-csi-provisioner_rbac.authorization.k8s.io_v1_clusterrole.yaml"
+          "topolvm-csi-provisioner_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml"
+          "topolvm-csi-provisioner_rbac.authorization.k8s.io_v1_role.yaml"
+          "topolvm-csi-provisioner_rbac.authorization.k8s.io_v1_rolebinding.yaml"
+          "topolvm-csi-resizer_rbac.authorization.k8s.io_v1_clusterrole.yaml"
+          "topolvm-csi-resizer_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml"
+          "topolvm-csi-resizer_rbac.authorization.k8s.io_v1_role.yaml"
+          "topolvm-csi-resizer_rbac.authorization.k8s.io_v1_rolebinding.yaml"
+          "topolvm-metrics_rbac.authorization.k8s.io_v1_role.yaml"
+          "topolvm-metrics_rbac.authorization.k8s.io_v1_rolebinding.yaml"
+          "topolvm-node-metrics_v1_service.yaml"
+          "topolvm-node_rbac.authorization.k8s.io_v1_clusterrole.yaml"
+          "topolvm-node_rbac.authorization.k8s.io_v1_clusterrolebinding.yaml"
+          "topolvm-node_v1_serviceaccount.yaml"
+          "topolvm.io_logicalvolumes.yaml"
+    )
+    for m in "${manifests[@]}"; do
+        cp -v "${src}/${m}" "${dest}/" || return 1
+        set_openshift_storage_ns "${dest}/${m}"
+    done
+}
+
+update_release_go() {
+    local src=$1
+    local arch=$2
+
+    case "$arch" in
+        amd64|x86_64)   release_file="${REPOROOT}/pkg/assets/release-x86_64.json"   ;;
+        arm64|aarch64)  release_file="${REPOROOT}/pkg/assets/release-aarch64.json"  ;;
+    esac
+
+    local staged_release
+    staged_release="$(mktemp -d)/$(basename "$release_file")"
+    cp "$release_file" "${staged_release}" || return 1
+    # shellcheck disable=SC2155
+    local images
+    images="$(parse_images "$src")" || return 1
+    cat images
+    # ensure that if a sed substitute fails, we bail out rather than clobber a release file
+    if ! (
+        while IFS=' ' read -r line; do
+          yq '.images['""']'
+        done <"$images"
+    ); then
+        return 1
+    fi
```

Comment:
All this construct looks a bit complex, cat images, if with nested while, etc. Can we please simplify this code?